embuilder.py

```diff
@@ -208,30 +208,28 @@ def main():
   if args.force:
     do_clear = True
 
+  system_libraries, system_tasks = get_system_tasks()
+
   # process tasks
   auto_tasks = False
-  tasks = args.targets
-  system_libraries, system_tasks = get_system_tasks()
-  if 'SYSTEM' in tasks:
-    tasks = system_tasks
-    auto_tasks = True
-  elif 'USER' in tasks:
-    tasks = PORTS
-    auto_tasks = True
-  elif 'MINIMAL' in tasks:
-    tasks = MINIMAL_TASKS
-    auto_tasks = True
-  elif 'MINIMAL_PIC' in tasks:
-    tasks = MINIMAL_PIC_TASKS
-    auto_tasks = True
-  elif 'ALL' in tasks:
-    tasks = system_tasks + PORTS
-    auto_tasks = True
+  tasks = set(args.targets)
```

Comment:
Unlike dict, I think python sets are still unordered by default or maybe use `OrderedSet` here?

Reply:
I'm not a python expert, always thought both `dict` and `set` are unordered though.

Current implementation has no sorting on the targets specified as command-line arguments, so I would stick this existing behaviour.

Reply:
Recent versions of python now guarantee insertion order for dicts: https://mail.python.org/pipermail/python-dev/2017-December/151283.html 

But for some reason they didn't do it for set.

The problem with just using `set()` here is that it doesn't guarantee insertion order, so when we go it iterate through the set later it could be in any order (unless you use `OrderedSet`.. which would keep the existing behaviour I think).